src/memjs/memjs.ts

```diff
@@ -0,0 +1,1777 @@
+      return new Server(
+        hostPort[0],
+        parseInt(hostPort[1] || "11211", 10),
+        userPass[0],
+        userPass[1],
+        options
+      );
+    });
+    return new Client(servers, options as any);
+  }
+
+  /**
+   * Given a serverKey fromlookupKeyToServerKey, return the corresponding Server instance
+   *
+   * @param  {string} serverKey
+   * @returns {Server}
+   */
+  serverKeyToServer(serverKey: string) {
+    return this.serverMap[serverKey];
+  }
+
+  /**
+   * Given a key to look up in memcache, return a serverKey (based on some
+   * hashing function) which can be used to index this.serverMap
+   *
+   * @param  {string} key
+   * * @returns {string}
+   */
+  lookupKeyToServerKey(key: string) {
+    return this.options.keyToServerHashFunction(this.serverKeys, key);
+  }
+
+  // ## Memcache Commands
+  //
+  // All commands return their results through a callback passed as the last
+  // required argument (some commands, like `Client#set`, take optional arguments
+  // after the callback).
+  //
+  // The callback signature always follows:
+  //
+  //     callback(err, [arg1[, arg2[, arg3[...]]]])
+  //
+  // In case of an error the _err_ argument will be non-null and contain the
+  // `Error`. A notable exception includes a `Client#get` on a key that doesn't
+  // exist. In this case, _err_ will be null, as will the _value and _extras_
+  // arguments.
+
+  /**
+   * GET
+   *
+   * Retrieves the value at the given key in memcache.
+   *
+   * The callback signature is:
+   *
+   *     callback(err, value, flags)
+   *
+   * _value_ and _flags_ are both `Buffer`s. If the key is not found, the
+   * callback is invoked with null for both arguments and no error
+   * @param key
+   * @param callback
+   */
+  get(key: string): Promise<{ value: Value | null; flags: Extras | null }>;
+  get(
+    key: string,
+    callback: (
+      error: Error | null,
+      value: Value | null,
+      flags: Extras | null
+    ) => void
+  ): void;
+  get(
+    key: string,
+    callback?: (
+      error: Error | null,
+      value: Value | null,
+      flags: Extras | null
+    ) => void
+  ): Promise<{ value: Value | null; flags: Extras | null }> | void {
+    var self = this;
+    if (callback === undefined) {
+      return promisify(function (callback) {
+        self.get(key, function (err, value, flags) {
+          callback(err, { value: value, flags: flags });
+        });
+      });
+    }
+    var logger = this.options.logger;
+    this.incrSeq();
+    var request = makeRequestBuffer(constants.OP_GET, key, "", "", this.seq);
+    this.perform(key, request, this.seq, function (err, response) {
+      if (err) {
+        if (callback) {
+          callback(err, null, null);
+        }
+        return;
+      }
+      switch (response!.header.status) {
+        case constants.RESPONSE_STATUS_SUCCCESS:
+          if (callback) {
+            var deserialized = self.serializer.deserialize(
```

Comment:
surprised you didn't do a bulk find and replace for var -> const, and let vscode find all the ones that need to be lets